src/upgrades/mongodb_upgrade.py

```diff
@@ -419,6 +558,82 @@ def step_down_primary_and_wait_reelection(self) -> None:
                 if new_primary == old_primary:
                     raise FailedToElectNewPrimaryError()
 
+    def are_pre_upgrade_operations_config_server_successful(self):
+        """Runs pre-upgrade operations for config-server and returns True if successful."""
+        if not self.charm.is_role(Config.Role.CONFIG_SERVER):
+            return False
+
+        if not self.is_feature_compatibility_version(Config.Upgrade.FEATURE_VERSION_6):
+            logger.debug(
+                "Not all replicas have the expected feature compatibility: %s",
+                Config.Upgrade.FEATURE_VERSION_6,
+            )
+            return False
+
+        self.set_mongos_feature_compatibilty_version(Config.Upgrade.FEATURE_VERSION_6)
```

Comment:
We could do it there too! But I think it would be better to keep all the upgrade stuff together. Do you see a substantial benefit to moving it to the start of the config server?

The start hook handles a LOT of functionality already and I would prefer to keep that to the bare essentials at this point.

But I am open to moving it i there is a substantial benefit!

Reply:
I don't have a strong opinion about it, but my idea was mainly that this doesn't need to be set multiple times or during upgrades specifically. You can ignore